core/src/test/java/com/graphhopper/routing/util/DataFlagEncoderTest.java

```diff
@@ -335,23 +336,23 @@ public void stringToMeterException() {
 
     @Test
     public void stringToTons() {
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("1.5"), DELTA);
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("1.5 t"), DELTA);
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("1.5   t"), DELTA);
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("1.5 tons"), DELTA);
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("1.5 ton"), DELTA);
-        assertEquals(1.5, OSMMaxWeightParser.stringToTons("3306.9 lbs"), DELTA);
-        assertEquals(3, OSMMaxWeightParser.stringToTons("3 T"), DELTA);
-        assertEquals(3, OSMMaxWeightParser.stringToTons("3ton"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("1.5"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("1.5 t"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("1.5   t"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("1.5 tons"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("1.5 ton"), DELTA);
+        assertEquals(1.5, OSMWeightExtractor.stringToTons("3306.9 lbs"), DELTA);
+        assertEquals(3, OSMWeightExtractor.stringToTons("3 T"), DELTA);
+        assertEquals(3, OSMWeightExtractor.stringToTons("3ton"), DELTA);
 
         // maximum gross weight
-        assertEquals(6, OSMMaxWeightParser.stringToTons("6t mgw"), DELTA);
+        assertEquals(6, OSMWeightExtractor.stringToTons("6t mgw"), DELTA);
     }
 
     @Test(expected = NumberFormatException.class)
     public void stringToTonsException() {
         // Unexpected values
-        OSMMaxWeightParser.stringToTons("weight limit 1.5t");
+        OSMWeightExtractor.stringToTons("weight limit 1.5t");
```

Comment:
As there is now a dedicated class for this, would you mind to move the related tests into a new OSMWeightExtractorTest class?

Reply:
Ah yes. Didn't really think that the values might have special cases.

btw: we should think about optional tests using the live json data provided by taginfo: https://taginfo.openstreetmap.org/api/4/key/values?key=maxweight&filter=all&lang=de&sortname=count&sortorder=desc&rp=27&qtype=value&format=json_pretty

It should be pretty easy to automate this and get an idea how big the coverage of our parsers is.